File: live-test/typescript/src/generic_super.ts
// Generic super fixture: Repo<T>.find overrides abstract BaseRepo<T>.find —
// super resolution must see through the generic base.
export abstract class BaseRepo<T> {
    abstract find(id: number): T;
}

export class Repo<T> extends BaseRepo<T> {
    find(id: number): T {
        throw new Error("not implemented");
    }
}

File: live-test/python/src/generic_super.py
# Super-method fixture: Repo.find must resolve to BaseRepo.find through the
# parameterized base BaseRepo[T] — generics must not hide the override edge.
"""Generic[T] class with overridden method."""
from typing import Generic, TypeVar

T = TypeVar("T")


class BaseRepo(Generic[T]):
    def find(self, id: int) -> T:
        raise NotImplementedError


class Repo(BaseRepo[T], Generic[T]):
    def find(self, id: int) -> T:
        raise NotImplementedError
